components/Editor/index.tsx

```diff
@@ -238,6 +238,83 @@ const Editor = ({ readOnly = false }: Props) => {
   ]
   useRegisterActions(actions, [highlightCode])
 
+  const handleCommentLine = useCallback(() => {
+    if (!editorRef.current) {
+      return
+    }
+    const textareaRef = editorRef.current._input
+    const lineNumberStart = textareaRef.value
+      .substring(0, textareaRef.selectionStart)
+      .split('\n').length
+    const lineNumberEnd = textareaRef.value
+      .substring(0, textareaRef.selectionEnd)
+      .split('\n').length
+
+    const selectionStart = textareaRef.selectionStart
+    const selectionEnd = textareaRef.selectionEnd
+    const lines = cairoCode.split('\n')
+    const linesToHandle: number[] = []
+    for (let k = lineNumberStart; k <= lineNumberEnd; k++) {
+      linesToHandle.push(k)
+    }
+
+    function isCommentLine(input: string) {
```

Comment:
would be good to move this function outside of the callback, so that it doesn't need to be recreated each time the callback is called, for better performance